src/leader.py

```diff
@@ -152,6 +158,77 @@ def lead(self):
             self.turn(code, position)
 
 
+    def drive_in_circle(self):
+        """Drives in a circle."""
+        self.bw.speed(MAX_SPEED)
+        self.bw.forward()
+
+        self.turn_left()
+
+
+    def drive_wide_winding(self):
+        """Drives in a wide, snake-like pattern."""
+        self.demo_stop_flag = False
+        self.bw.speed(MAX_SPEED)
+        self.bw.forward()
+
+        self.turn_right()
+        self.demo_turn_wait()
+
+        if self.demo_stop_flag: return
+
+        self.turn_left()
+        self.demo_turn_wait()
+
+    def drive_winding(self):
+        """Drives in a snake-like pattern."""
+        self.demo_stop_flag = False
+        self.bw.speed(MAX_SPEED)
+        self.bw.forward()
+
+        self.fw.turn(110)
+        self.demo_turn_wait()
+
+        if self.demo_stop_flag: return
+
+        self.fw.turn(70)
+        self.demo_turn_wait()
+
+
+    def stop_demo_check(self):
+        """
+        Checks if the demo should stop.
+
+        Changes the demo_stop_flag based on any controller input recieved for
+        .1 consecutive seconds. If any button has been pressed for over .1
```

Comment:
What? Why would we do this? Why not just end the demo as soon as any input is recorded?

Reply:
As soon as the demo is started it will read a controller input button pressed to start the demo, and most likely read it again, and if any button press terminates the demo it will instantly start and terminate.